src/job_cache/job_cache.cpp

```diff
@@ -1140,7 +1140,9 @@ wcl::optional<MatchingJob> Cache::read(const FindJobRequest &find_request) {
 
       // Finally copy the file (as efficently as we can) to
       // the destination.
-      copy_or_reflink(tmp_file.c_str(), pair.first.c_str(), mode);
+      std::string tmp_dst = pair.first + "." + rng.unique_name();
+      copy_or_reflink(tmp_file.c_str(), tmp_dst.c_str(), mode);
+      rename_no_fail(tmp_dst.c_str(), pair.first.c_str());
```

Comment:
nit: we should write a single function for this since its just a common pattern `rename_over_no_fail()` or something

Reply:
The problem is that the thing you want to rename changes, and you need access to a fast source of high entropy rng to do it so the interface is kinda strange.

Reply:
Yeah, its unclear. The dep on rng makes it a mucky api. Its at least worth looking at since we expect this to continuously bite us.

I'm willing to take the API tradeoff _if_ we can avoid an entire class of bugs